rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/service/IdamProfileService.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.hmcts.ccd.definition.store.rest.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.ccd.definition.store.domain.ApplicationParams;
+import uk.gov.hmcts.ccd.definition.store.repository.SecurityUtils;
+import uk.gov.hmcts.ccd.definition.store.rest.model.IDAMProperties;
+import uk.gov.hmcts.reform.auth.checker.spring.serviceanduser.ServiceAndUserDetails;
+
+import static org.springframework.http.HttpMethod.GET;
+
+@Service
+public class IdamProfileService {
```

Comment:
Why not to name it IdamProfileClient and move it to 'client' package. If we would do it to all classes that use RestTemplate it would be easy to find service dependencies for integration. Now we have client code mangled with services. Just a thought. This could be a separate task.